python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -815,7 +815,7 @@ def pid(self):
     cluster.head_node.kill_gcs_server(False)
 
     print("Start gcs")
-    sleep(2)
```

Comment:
why?

Reply:
I found that this test, `test_gcs_fault_tolerance`, only passes when the timeout here is extended. However, doing so exceeds the 300-second limit for Bazel medium-size tests, so I changed the size to large as well.

Reply:
Why this is working before? Any change from this PR makes us to increase the sleep?

Reply:
I found that it is because of this line

https://github.com/ray-project/ray/blob/6a3e35c75280df0a82e0013ff3c9861435f746a7/src/ray/gcs/redis_context.cc#L720